zomes/rea_economic_resource/storage/src/lib.rs

```diff
@@ -135,43 +135,70 @@ impl TryFrom<CreationPayload> for EntryData {
         let raise_action = get_builtin_action("raise").unwrap();
         let lower_action = get_builtin_action("lower").unwrap();
         let action_id = String::from(e.get_action());
+        let maybe_resource_specification = match conforming.clone() {
+            Some(conforms_to_spec) => Some(get_resource_specification(conforms_to_spec)?),
+            None => None,
+        };
+        let default_unit_of_resource = match maybe_resource_specification.clone() {
+            Some(ResourceSpecificationResponse { default_unit_of_resource, .. }) => default_unit_of_resource,
+            None => None
+        };
+        // first choice is the unit passed in on the event
+        // value, fallback is the default_unit_of_resource
+        // set on the resource specification
+        let quantity_value = match e.resource_quantity.to_owned() {
+            MaybeUndefined::Some(resource_quantity) => {
+                let unit = match resource_quantity.get_unit() {
+                    Some(unit) => Some(unit),
+                    None => default_unit_of_resource
+                };
+                Some(QuantityValue::new(resource_quantity.get_numerical_value(), unit))
+            },
+            _ => None,
+        };
+        // capture the unit_of_effort from any default
+        // set on the Resource Specification this conforms to,
+        // if any
+        let unit_of_effort = match maybe_resource_specification {
```

Comment:
I think there's an error in this logic. `EconomicResource.UnitOfEffort` should come from the `EconomicEvent.effortQuantity.hasUnit` if set and fallback to the one from the associated `ResourceSpecification.defaultUnitOfEffort`, correct @fosterlynn?

That being the case I think we can also make this a bit more efficient by only querying the `ResourceSpecification` if the necessary fields aren't set in the `EconomicEvent`.

Reply:
sounds right. K

Reply:
> EconomicResource.UnitOfEffort should come from the EconomicEvent.effortQuantity.hasUnit if set and fallback to the one from the associated ResourceSpecification.defaultUnitOfEffort, correct @fosterlynn?

Well, it is hard to imagine an event that creates a resource having an effortQuantity.  So it should take the one from the resource specification as you suggested, if it is there, when created.  I also think we need the capability to set it directly as part of creating or updating a resource, which would take precedence over the one in the resource specification.  I also don't see a big problem with doing what you suggest with the effortQuantity.hasUnit if it is not already set.  (Not sure if we want to overwrite it in that situation, but could perhaps be persuaded. It seems like a user choice, and I'm actually not sure how many user groups would want it to be a firm setting, or just a default that can be overridden on particular events.)

Reply:
"So it should take the one from the resource specification as you suggested, if it is there, when created." this bit we already do